database.js

```diff
@@ -22,8 +22,25 @@ const getBookById = (book_id) => {
   return database.one(sql, [book_id])
 }
 
+const findBooks = (options) => {
+  const sql = 'SELECT * FROM books WHERE LOWER(title) LIKE $1 OR LOWER(description) LIKE $1 OR LOWER(author) LIKE $1'
+  let variables = []
+  if (options.q) {
+    let q = options.q
+      .toLowerCase()
+      .replace(/^ */, '%')
+      .replace(/ *$/, '%')
+      .replace(/ +/g, '%')
+
+    variables.push(q)
+
+  return database.any(sql, variables)
```

Comment:
This doesn't work.  The code is incorrectly aligned which is helping to hide the issue, but you're only creating a database query if `options` are provided.  If `options` are always provided, then you should omit the `if` statement, and just create the variables array with the `q` value directly...